src/adapters.ts

```diff
@@ -142,3 +142,57 @@ export function redisCacheAdapter(redisCache: RedisLikeCache): Cache {
     },
   };
 }
+
+export interface RedisJsonLikeCache {
+  name?: string;
+  json: {
+    set(
+      key: string,
+      path: string,
+      value: Record<string, any>,
+    ): Promise<string | null>;
+    get(key: string): Promise<string | null>;
+    del(key: string): Promise<unknown>;
+  };
+  expireAt(key: string, timestamp: number): Promise<unknown>;
+}
+
+export function redisJsonCacheAdapter(
+  redisJsonCache: RedisJsonLikeCache,
+): Cache {
+  return {
+    name: redisJsonCache.name || 'RedisJson',
+    delete(key) {
+      return redisJsonCache.json.del(key);
+    },
+    async set(key: string, value: Record<string, any>) {
```

Comment:
We should be able to remove the explicit typing here and go with `set(key, value)` that way we have type-safety on things like `value?.metadata`